.taskcluster.yml

```diff
@@ -6,185 +6,173 @@ version: 1
 policy:
   pullRequests: public
 tasks:
-####################################################################################################
-# Task: Pull requests
-####################################################################################################
-  - $if: 'tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "edited", "synchronize"]'
-    then:
-      $let:
-        decision_task_id: {$eval: as_slugid("decision_task")}
-        expires_in: {$fromNow: '1 year'}
-        scheduler_id: taskcluster-github
-        user: ${event.sender.login}
-        head_branch: ${event.pull_request.head.ref}
-        head_rev: ${event.pull_request.head.sha}
-        repository: ${event.pull_request.head.repo.clone_url}
-        pull_request_title: ${event.pull_request.title}
-        pull_request_number: ${event.pull_request.number}
-        pull_request_url: ${event.pull_request.html_url}
-      in:
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    scheduler_id: taskcluster-github
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.ref}
+        else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    worker_type:
+      $if: 'tasks_for in ["github-push", "github-release"] && event.repository.html_url[:34] == "https://github.com/mozilla-mobile/"'
```

Comment:
I think having `worker_type` at the top makes things simpler => You need 5 lines to handles all cases. Otherwise, here's how it would look like:

```yaml
# In PR
$mergeDeep:
  # [...]
  - worker-type: mobile-1-decision

# In pushes
$mergeDeep:
  # [...]
  - worker-type:
    $if: 'event.repository.html_url == "https://github.com/mozilla-mobile/android-components"'
      then: mobile-3-decision
      else: mobile-1-decision

# In releases
$mergeDeep:
  # [...]
  - worker-type:
    $if: 'event.repository.html_url == "https://github.com/mozilla-mobile/android-components"'
      then: mobile-3-decision
      else: mobile-1-decision
```

=> You'd get 9 lines and you'd duplicate the logic + the URL twice. What do you think?

Reply:
You could have a `$let` at the top of each `task` (e.g.: [here](https://github.com/mozilla-mobile/android-components/pull/1588/files#diff-ac0229d1171c0983b27003af4c7441a5R90)). I think there's be a greater line count, but I think it'll be worth it because the conditional will be easier to parse (and the logic will be closer to where it's applied)